openassessment/xblock/openassessmentblock.py

```diff
@@ -524,6 +524,46 @@ def get_student_item_dict(self, anonymous_user_id=None):
         }
         return student_item_dict
 
+    @togglable_mobile_support
+    def author_view(self, context=None):  # pylint: disable=unused-argument
+        """The main view of OpenAssessmentBlock, displayed when viewing courses.
+
+        View which displays the legacy UI for authoring in Studio.
+
+        Args:
+            context: Not used for this view.
+
+        Returns:
+            (Fragment): The HTML Fragment for this XBlock, which determines the
+            general frame of the Open Ended Assessment Question.
+        """
+        # On page load, update the workflow status.
+        # We need to do this here because peers may have graded us, in which
+        # case we may have a score available.
+
+        try:
+            self.update_workflow_status()
+        except AssessmentWorkflowError:
+            # Log the exception, but continue loading the page
+            logger.exception('An error occurred while updating the workflow on page load.')
+
+        ui_models = self._create_ui_models()
+        # All data we intend to pass to the front end.
+        context_dict = {
+            "mfe_views": False,
+            "title": self.title,
+            "prompts": self.prompts,
+            "prompts_type": self.prompts_type,
+            "rubric_assessments": ui_models,
+            "show_staff_area": self.is_course_staff and not self.in_studio_preview,
+        }
+        if any(model.name == "leaderboard" for model in ui_models):
+            context_dict["leaderboard_ui_model"] = [
+                model for model in ui_models if model.name == "leaderboard"
+            ][0]
```

Comment:
Slightly different version that I think (1) reads better and (2) pops duplicate leaderboard from `ui_models`

```suggestion
        leaderboard_step = None

        for step_number, assessment_step in enumerate(ui_models):
            if step.name == "leaderboard"
                leaderboard_step = ui_models.pop(step_number)
                break
            
        context_dict = {
            "mfe_views": False,
            "title": self.title,
            "prompts": self.prompts,
            "prompts_type": self.prompts_type,
            "rubric_assessments": ui_models,
            "leaderboard_step": leaderboard_step,
            "show_staff_area": self.is_course_staff and not self.in_studio_preview,
        }

```

Reply:
FWIW, my vote would also be to have a shared function to generate the context for these, but with `mfe_views` just forced off for studio.